static/js/muting_ui.js

```diff
@@ -207,3 +210,20 @@ export function handle_user_updates(muted_user_ids) {
     muting.set_muted_users(muted_user_ids);
     rerender_for_muted_user();
 }
+
+function update_topic_recipient_bar(stream_id, topic, row) {
+    if (row !== undefined && row.length !== 0) {
+        const message_id = rows.id_for_recipient_row(row);
+        const message = message_lists.current.get(message_id);
+        message_lists.current.view._rerender_header([{msg: message}]);
+    } else {
+        const message_groups = message_lists.current.view._message_groups;
+        const message_group = message_groups.find(
+            (message_group) =>
+                message_group.stream_id === stream_id && message_group.topic === topic,
+        );
+        if (message_group !== undefined) {
+            message_lists.current.view._rerender_header(message_group.message_containers);
+        }
+    }
+}
```

Comment:
This function is wrong; and I think the fix involves no longer passing `row` here.  The bugs are (1) that we can have more than one message list rendered at a time that might need updating and (2) we might have multiple recipient bars for a given topic that need updating (E.g. in a search or stream-level narrow where there's interleaved messages from the topic in question).

I think the existing call to `rerender_on_topic_update` should handle what we need here.